extensions/wikia/GlobalNavigation/scripts/GlobalNavigationAccountNavigation.js

```diff
@@ -87,5 +84,38 @@ require([
 					scrollFix.restoreScrollY($globalNavigation);
 				});
 		}
+	}
+
+	$(function () {
+		var $entryPoint, $authEntryPoints;
+
+		$entryPoint = $('#AccountNavigation');
+
+		if (!win.wgUserName && $entryPoint.hasClass('newAuth')) {
+			$authEntryPoints = $('.auth-link.register, .auth-link.sign-in, a.sign-in');
+
+			$authEntryPoints.click(function (event) {
+				if (event.which !== 1 || event.shiftKey || event.altKey || event.metaKey || event.ctrlKey) {
+					return;
+				}
+				if (event) {
+					event.preventDefault();
+					event.stopPropagation();
+				}
+
+				if (event.target.classList.contains('register')) {
+					require(['AuthModal'], function (authModal) {
+						authModal.register();
+					});
+				} else {
+					require(['AuthModal'], function (authModal) {
+						authModal.login();
+					});
+				}
+			});
```

Comment:
what do you think about extracting code responsible for new modal to separate function?